tests/functional/olp-cpp-sdk-dataservice-read/VolatileLayerClientTest.cpp

```diff
@@ -0,0 +1,249 @@
+
+  {
+    SCOPED_TRACE("Get Partitions Invalid Layer Test");
+
+    VolatileLayerClient client(hrn, "InvalidLayer", settings_);
+
+    olp::client::Condition condition(kTimeout);
+    olp::dataservice::read::VolatileLayerClient::PartitionsResponse
+        partitions_response;
+
+    auto callback =
+        [&](olp::dataservice::read::VolatileLayerClient::PartitionsResponse
+                response) {
+          partitions_response = std::move(response);
+          condition.Notify();
+        };
+    client.GetPartitions(olp::dataservice::read::PartitionsRequest(),
+                         std::move(callback));
+    ASSERT_TRUE(condition.Wait());
+    EXPECT_FALSE(partitions_response.IsSuccessful());
+  }
+
+  {
+    SCOPED_TRACE("Get Partitions Invalid HRN Test");
+
+    VolatileLayerClient client(olp::client::HRN("Invalid"), GetTestLayer(),
+                               settings_);
+
+    olp::client::Condition condition(kTimeout);
+    olp::dataservice::read::VolatileLayerClient::PartitionsResponse
+        partitions_response;
+
+    auto callback =
+        [&](olp::dataservice::read::VolatileLayerClient::PartitionsResponse
+                response) {
+          partitions_response = std::move(response);
+          condition.Notify();
+        };
+    client.GetPartitions(olp::dataservice::read::PartitionsRequest(),
+                         std::move(callback));
+    ASSERT_TRUE(condition.Wait());
+    EXPECT_FALSE(partitions_response.IsSuccessful());
+  }
+}
+
+TEST_P(VolatileLayerClientTest, GetPartitionsDifferentFethOptions) {
+  olp::client::HRN hrn(GetTestCatalog());
+  {
+    SCOPED_TRACE("Get Partitions Online Only");
+
+    VolatileLayerClient client(hrn, GetTestLayer(), settings_);
+
+    olp::client::Condition condition(kTimeout);
+    olp::dataservice::read::VolatileLayerClient::PartitionsResponse
+        partitions_response;
+
+    auto callback =
+        [&](olp::dataservice::read::VolatileLayerClient::PartitionsResponse
+                response) {
+          partitions_response = std::move(response);
+          condition.Notify();
+        };
+    client.GetPartitions(
+        PartitionsRequest().WithFetchOption(FetchOptions::OnlineOnly),
+        std::move(callback));
+    ASSERT_TRUE(condition.Wait());
+    EXPECT_TRUE(partitions_response.IsSuccessful());
+  }
+  {
+    SCOPED_TRACE("Get Partitions Cache Only");
+
+    VolatileLayerClient client(hrn, GetTestLayer(), settings_);
+
+    olp::client::Condition condition(kTimeout);
+    olp::dataservice::read::VolatileLayerClient::PartitionsResponse
+        partitions_response;
+
+    auto callback =
+        [&](olp::dataservice::read::VolatileLayerClient::PartitionsResponse
+                response) {
+          partitions_response = std::move(response);
+          condition.Notify();
+        };
+    client.GetPartitions(
+        PartitionsRequest().WithFetchOption(FetchOptions::CacheOnly),
+        std::move(callback));
+    ASSERT_TRUE(condition.Wait());
```

Comment:
Please, rebase your change: timeout for condition will be passed in Wait(...) instead of constructor.